src/main.c

```diff
@@ -703,7 +650,11 @@ print_choices(int selection)
 	if ((line = calloc(sizeof(*line), line_length)) == NULL)
 		err(1, "calloc");
 
-	SLIST_FOREACH(choice, choices, choices) {
+	for (i = 0;
+	     i < (ssize_t)choices.nmemb
+	     && i < lines
```

Comment:
Great catch, fixed.

On Mon, Jan 04, 2016 at 01:36:56AM -0800, Calle Erlandsson wrote:
> > @@ -703,7 +650,11 @@ print_choices(int selection)
> >  	if ((line = calloc(sizeof(*line), line_length)) == NULL)
> >  		err(1, "calloc");
> >  
> > -	SLIST_FOREACH(choice, choices, choices) {
> > +	for (i = 0;
> > +	     i < (ssize_t)choices.nmemb
> > +	     && i < lines
> 
> I think there might be an off by one error here. Running pick with more choices than would fit on screen, the selection is invisible. I think this is because the last choice printed to the terminal makes it scroll and then the first choice is overwritten by the query line.
> 
> Changing this to `&& i < lines + 1` locally fixes the issue for me.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thoughtbot/pick/pull/82/files#r48715092

-- 
:wq